hooks/proprietary/PSMDST.js

```diff
@@ -17,22 +17,26 @@
 'use strict'
 
 /*
-0  1  2  3
-|  |  |  |
-$PSMDST,xx,yy,nn*CS
+0  1  2  3  4
+|  |  |  |  |
+$PSMDST,Z,xx,yy,nn*CS
 where:
 PSMDST     	Raymarine Seatalk1 datagram sentence
-0 			00-9C       	Datagram type
-1 			hex       	First datagram content
-2 			hex   		Last datagram content
-3 			hex      	Checksum
+0       C/R       R for Recevied messages, C for sent messages *Note: This field only exists in later firmware versions of the ShipModul Miniplex
+1 			00-9C     Datagram type
+2 			hex       First datagram content
+3 			hex   		Last datagram content
+4 			hex      	Checksum
 */
 
 const seatalkHooks = require('../seatalk')
 
 module.exports = function (input, session) {
   const { id, sentence, parts, tags } = input
-  const key = '0x' + parts[0].toUpperCase()
+  if (parts[0].toUpperCase() === 'R') {
```

Comment:
I think we should structure the code so that
- if there is no extra field parse as before this PR - test on the number of fields
- if there is the extra field
  - if it is something else return undefined so that no output is generated
  - if it is 'R' remove it and parse as before this PR